app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -751,6 +757,12 @@ class HomeFragment : Fragment() {
         lifecycleScope.launch(IO) {
             requireComponents.reviewPromptController.promptReview(requireActivity())
         }
+
+        if (FeatureFlags.showWallpapers) {
+            binding.wordmark.setOnClickListener {
+                requireComponents.wallpaperManager.switchToNextWallpaper(wallpaperContainer = binding.homeLayout)
```

Comment:
Rather than taking in the container, I think we should just return the next wallpaper from the manager, and let the container do what it needs to do with it.

Reply:
I think this is fine for the moment, as we don't have any other use case for `switchToNextWallpaper`, remember that this is just a prototype no the final version, this will be iterate over multiple times.

Reply:
It's very likely that we will remove this code

Reply:
I agree with @gabrielluong here. Generally speaking, I try keep as many Android components out of my classes as possible. It has several benefits such as making classes more unit testable and avoiding retaining references to large and expensive objects like a `Context`.

Even though this code is intended to be iterated on in the near future, I think we should try to ensure that any code we ship to users is production ready. Someone might download this nightly and then not updated for several weeks, for example.